app/src/main/java/com/x8bit/bitwarden/data/platform/datasource/network/interceptor/ResponseJsonKeyTransformerInterceptor.kt

```diff
@@ -0,0 +1,97 @@
+package com.x8bit.bitwarden.data.platform.datasource.network.interceptor
+
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonArray
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.buildJsonObject
+import okhttp3.Interceptor
+import okhttp3.Response
+import okhttp3.ResponseBody.Companion.toResponseBody
+
+/**
+ * An OkHttp interceptor that transforms the JSON response body by converting all JSON object keys
+ * to lowercase.
+ *
+ * This interceptor is useful for handling inconsistencies in JSON responses where key casing might
+ * vary.
+ */
+class ResponseJsonKeyTransformerInterceptor : Interceptor {
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+        val response = chain.proceed(request)
+
+        val responseBody = response.body
+            ?: return response
+
+        // Check if the response body is JSON. Return the original response if not.
+        val contentType = responseBody.contentType()
+            ?.takeIf { it.subtype.equals("json", ignoreCase = true) }
+            ?: return response
+
+        // Read the response body as a string.
+        val source = responseBody.source()
+        source.request(Long.MAX_VALUE)
+        val responseBodyString = source
+            .buffer
+            .readString(Charsets.UTF_8)
+            .takeUnless { it.isEmpty() }
+            ?: return response
+
+        // Transform the JSON object keys to lowercase.
+        val transformedJsonResponseBody = Json
+            .parseToJsonElementOrNull(responseBodyString)
+            ?.jsonObjectOrNull()
+            ?.lowercaseKeys()
+            ?.toString()
+            ?.toResponseBody(contentType)
+            ?: return response
+
+        return response.newBuilder()
+            .body(transformedJsonResponseBody)
+            .build()
+    }
+}
+
+/**
+ * Safely parses a JSON string to a JsonElement, returning null if the parsing fails.
+ */
+private fun Json.parseToJsonElementOrNull(json: String): JsonElement? = try {
+    parseToJsonElement(json)
+} catch (_: SerializationException) {
+    null
+}
+
+/**
+ * Safely casts a JsonElement to a JsonObject, returning null if the cast fails.
+ */
+private fun JsonElement.jsonObjectOrNull(): JsonObject? = this as? JsonObject
+
+/**
+ * Recursively converts all keys within a JsonObject and its nested objects/arrays to lowercase.
+ *
+ * @return A new JsonObject with all keys in lowercase.
+ */
+private fun JsonObject.lowercaseKeys(): JsonObject {
+    return buildJsonObject {
+        this@lowercaseKeys.entries
+            .forEach { (key: String, value: JsonElement) ->
+                val lowercaseKey = key.lowercase()
```

Comment:
Instead of lowercase can we use camelcase.

iOS has logic for that [here](https://github.com/bitwarden/ios/blob/edbe5f6b965fd2c89a917590f24de49690168038/BitwardenShared/Core/Platform/Services/API/Extensions/JSONDecoder%2BBitwarden.swift#L59), though I don't think we need to deal with snake case.

Reply:
Can we also document this in the readme, since we are not ignoring the server spec and using our own casing.

Reply:
Keys will not be transformed to camel case. Will update documentation appropriately. 👍